fol/constitutive_material_models/base.py

```diff
@@ -0,0 +1,42 @@
+from abc import ABC, abstractmethod
+from typing import Optional, Any, Tuple
+import jax.numpy as jnp
+from jax import Array
+
+class BaseConstitutiveModel(ABC):
+    """
+    Abstract base class for constitutive models.
+    Minimal interface - subclasses define their own evaluate signature.
+    """
+    
+    @abstractmethod
+    def evaluate(self, *args, **kwargs) -> Tuple:
+        """
+        Evaluate constitutive relation.
+        
+        Returns vary by material type:
+        - Hyperelastic: (energy, stress, tangent)
+        - Small-strain plastic: (stress, new_state)
+        - Large-strain plastic: (stress, new_state)
+        """
+        pass
+    
+
+
+
+class PlasticityModel(BaseConstitutiveModel):
```

Comment:
When you get a chance, could you move the PlasticityModel class definition to plasticity.py? That way we can keep the base class purely abstract in one place.
It would also be great if you could add your authorship information at the top of the file so we have proper attribution.